ambry-server/src/main/java/com/github/ambry/server/RepairRequestsSender.java

```diff
@@ -0,0 +1,237 @@
+    partitionIds = new ArrayList<>();
+    partition2Replicas = new HashMap<>();
+    for (ReplicaId replica : clusterMap.getReplicaIds(nodeId)) {
+      long partition = replica.getPartitionId().getId();
+      partitionIds.add(partition);
+      partition2Replicas.put(partition, replica);
+    }
+  }
+
+  public void run() {
+    List<RequestInfo> requestInfos = null;
+
+    while (!shutdown) {
+      try {
+        // sleep for some time
+        Thread.sleep(SLEEP_TIME_MS);
+
+        // loop all the partitions
+        for (long partitionId : partitionIds) {
+          if (shutdown) {
+            return;
+          }
+
+          // get the repair requests for this partition
+          requestInfos = getAmbryRequest(partitionId);
+          if (requestInfos == null || requestInfos.isEmpty()) {
+            continue;
+          }
+
+          // send the repair requests to the handler and wait for all the responses.
+          Set<Integer> requestsToDrop = new HashSet<>();
+          int inflightReqCount = requestInfos.size();
+          logger.info("RepairRequests Sender: {} sending {} repair requests ", nodeId, inflightReqCount);
+          List<ResponseInfo> responses = client.sendAndPoll(requestInfos, requestsToDrop, POLL_TIMEOUT_MS);
+          requestInfos.clear();
+
+          for (ResponseInfo info : responses) {
+            inflightReqCount--;
+            ReplicateBlobResponse res = (ReplicateBlobResponse) info.getResponse();
+            ReplicateBlobRequest req = (ReplicateBlobRequest) info.getRequestInfo().getRequest();
+            if (res.getError() == ServerErrorCode.No_Error) {
+              RepairRequestRecord.OperationType type;
+              if (req.getOperationType() == RequestOrResponseType.TtlUpdateRequest) {
+                type = RepairRequestRecord.OperationType.TtlUpdateRequest;
+              } else {
+                type = RepairRequestRecord.OperationType.DeleteRequest;
+              }
+              db.removeRepairRequests(req.getBlobId().toString(), type);
+              logger.info("RepairRequests Sender: Repaired {}", req);
+            } else {
+              // LOCAL_CONSISTENCY_TODO: metrics for all the errors happens in RepairRequestsSender.
+              // Ideally we may need create alert if we cannot repair the requests in timely matter.
+              logger.info("RepairRequests Sender: failed to repair {} response {} ", req, res);
+            }
+          }
+
+          if (inflightReqCount > 0) {
+            logger.error("RepairRequests Sender: timeout waiting for repairs {} {}", nodeId, inflightReqCount);
+          }
+        } // loop all the partitions
+      } catch (Throwable e) {
+        // TODO add metric to track background threads
+        logger.error("RepairRequests Sender: Exception when handling request", e);
+        // this is bad and we need to shutdown the app
+        Runtime.getRuntime().halt(1);
```

Comment:
we don't have to kill the entire process just because the exception from repair.

Reply:
Remove the halt.